nano/node/active_transactions.cpp

```diff
@@ -67,99 +67,152 @@ void nano::active_transactions::stop ()
 
 void nano::active_transactions::block_cemented_callback (std::shared_ptr<nano::block> const & block_a)
 {
-	auto transaction = node.store.tx_begin_read ();
+	nano::store::read_transaction transaction = node.store.tx_begin_read ();
+	auto status_type = determine_election_status (transaction, block_a);
 
-	boost::optional<nano::election_status_type> election_status_type;
-	if (!confirmation_height_processor.is_processing_added_block (block_a->hash ()))
+	if (!status_type)
+		return;
+
+	if (*status_type == nano::election_status_type::inactive_confirmation_height)
 	{
-		election_status_type = confirm_block (transaction, block_a);
+		process_inactive_confirmation (transaction, block_a);
 	}
 	else
 	{
-		// This block was explicitly added to the confirmation height_processor
-		election_status_type = nano::election_status_type::active_confirmed_quorum;
+		process_active_confirmation (transaction, block_a, *status_type);
 	}
 
-	if (election_status_type.is_initialized ())
+	handle_final_votes_confirmation (block_a, transaction, *status_type);
+}
+
+boost::optional<nano::election_status_type> nano::active_transactions::determine_election_status (nano::store::read_transaction const & transaction, std::shared_ptr<nano::block> const & block)
+{
+	boost::optional<nano::election_status_type> status_type;
+
+	if (!confirmation_height_processor.is_processing_added_block (block->hash ()))
 	{
-		if (election_status_type == nano::election_status_type::inactive_confirmation_height)
-		{
-			nano::account account{};
-			nano::uint128_t amount (0);
-			bool is_state_send (false);
-			bool is_state_epoch (false);
-			nano::account pending_account{};
-			node.process_confirmed_data (transaction, block_a, block_a->hash (), account, amount, is_state_send, is_state_epoch, pending_account);
-			node.observers.blocks.notify (nano::election_status{ block_a, 0, 0, std::chrono::duration_cast<std::chrono::milliseconds> (std::chrono::system_clock::now ().time_since_epoch ()), std::chrono::duration_values<std::chrono::milliseconds>::zero (), 0, 1, 0, nano::election_status_type::inactive_confirmation_height }, {}, account, amount, is_state_send, is_state_epoch);
-		}
-		else
+		status_type = confirm_block (transaction, block);
+	}
+	else
+	{
+		status_type = nano::election_status_type::active_confirmed_quorum;
+	}
+
+	return status_type;
+}
+
+void nano::active_transactions::process_inactive_confirmation (nano::store::read_transaction const & transaction, std::shared_ptr<nano::block> const & block)
+{
+	nano::account account{};
```

Comment:
The {} is mostly redundant. For some types this forces the default constructor to be run, which could give it an initialized value versus an uninitialized one.